app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
+  def check
```

Comment:
The button says "Reindex User," but this appears to do more than that. I'm honestly not sure what the best solution is here -- rename the button or separate the actions. 

Renaming the button something more vague like "Troubleshoot Account" is certainly easiest, but it means running checks we don't necessarily need (although I don't think any of them are resource-intensive, so that's not a _terrible_ problem), and it's not particularly informative from an interface standpoint. 

Maybe the thing to do would be rename the button, add some text to the top of the page the button appears on saying "To do X, Y, and Z to this account, choose 'Button Name,'" and then include a note here in the controller to update the view if adding/removing functions.